src/main/java/cn/nukkit/entity/item/EntityMinecartEmpty.java

```diff
@@ -45,24 +68,418 @@ protected float getGravity() {
+							return true;
+						}
+						break;
+				}
+				break;
+			case Rail.SLOPED_ASCENDING_EAST:
+				switch(currentDirection) {
+					case Entity.EAST:
+						// Headed east up
+						diff = this.z - this.getFloorZ();
+						if (diff != 0 && diff <= .5) {
+							dz = (this.getFloorZ() - .1) - this.z;
+							this.move(0, 1, dz);
+							return true;
+						}
+						break;
+					case Entity.WEST:
+						// Headed west down
+						diff = this.z - this.getFloorZ();
+						if (diff != 0 && diff >= .5) {
+							dz = (this.getFloorZ() + 1) - this.z;
+							this.move(0, -1, dz);
+							return true;
+						}
+						break;
+				}
+				break;
+			case Rail.SLOPED_ASCENDING_WEST:
+				switch(currentDirection) {
+					case Entity.WEST:
+						// Headed west up
+						diff = this.z - this.getFloorZ();
+						if (diff != 0 && diff >= .5) {
+							dz = (this.getFloorZ() + 1) - this.z;
+							this.move(0, 1, dz);
+							return true;
+						}
+						break;
+					case Entity.EAST:
+						// Headed east down
+						diff = this.z - this.getFloorZ();
+						if (diff != 0 && diff <= .5) {
+							dz = (this.getFloorZ() - .1) - this.z;
+							this.move(0, -1, dz);
+							return true;
+						}
+						break;
+				}
+				break;
+		}
+		return false;
+    }
+	
+	/**
+	 * Move the minecart as long as it will still be moving on to another piece of rail.
+	 * @return boolean True if the minecart moved.
+	 */
+    @Override
+    private void moveIfRail() {
+		nextMoveVector = this.moveVector[this.direction];
+		nextMoveVector = nextMoveVector.multiply(this.moveSpeed);
+		newVector = this.add(nextMoveVector.x, nextMoveVector.y, nextMoveVector.z);
+		possibleRail = this.getCurrentRail();
+		if (possibleRail.getId(), {Block.RAIL, Block.ACTIVATOR_RAIL, Block.DETECTOR_RAIL, Block.POWERED_RAIL}) {
+			this.moveUsingVector(newVector);
+			return true;
+		}
+    }
+	
+	/**
+	 * Invoke the normal move code, but first need to convert the desired position vector into the
+	 * delta values from the current position.
+	 * @param Vector3 $desiredPosition
+	 */
+    @Override
+    private void moveUsingVector(Vector3 desiredPosition){
+		dx = desiredPosition.x - this.x;
+		dy = desiredPosition.y - this.y;
+		dz = desiredPosition.z - this.z;
+		this.move(dx, dy, dz);
+    }
+	
+    @Override
+    public void getNearestRail() {
+		minX = NukkitMath.floorFloat(this.boundingBox.minX);
+		minY = NukkitMath.floorFloat(this.boundingBox.minY);
+		minZ = NukkitMath.floorFloat(this.boundingBox.minZ);
+		maxX = NukkitMath.ceilFloat(this.boundingBox.maxX);
+		maxY = NukkitMath.ceilFloat(this.boundingBox.maxY);
+		maxZ = NukkitMath.ceilFloat(this.boundingBox.maxZ);
+		rails = [];
+		for (z = minZ; z <= maxZ; ++z) {
+			for (x = minX; x <= maxX; ++x) {
+				for (y = minY; y <= maxY; ++y) {
+					Block block = this.level.getBlock(this.temporalVector.setComponents(x, y, z));
+					if (block.getId(), {Block.RAIL, Block.ACTIVATOR_RAIL, Block.DETECTOR_RAIL, Block.POWERED_RAIL}) rails[] = block;
```

Comment:
Well I am not a web programmer. You should not code java in the way you code a web language.